### PR TITLE
K8S/Scopes: App-server for storing scope objects

### PR DESCRIPTION
**Pull request by kylebrandt**
**What is this feature?**

Experimental API/Crud for Scope objects.

### CODE REVIEW

File: pkg/apis/scope/v0alpha1/types.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+type Scope struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ScopeSpec `json:"spec,omitempty"`
+}
+
+type ScopeSpec struct {
+	// S does nothing and is a stub
+	S string `json:"s"`
```
Comment (ryantxu) [inline pkg/apis/scope/v0alpha1/types.go:17]: 🤣 yup -- this is my biggest question... what goes here

File: pkg/apis/peakq/v0alpha1/zz_generated.openapi.go
Comment (ryantxu) [inline pkg/apis/peakq/v0alpha1/zz_generated.openapi.go:1]: remove

File: pkg/apis/scope/v0alpha1/types.go (hunk 1 above)
Comment (bergquist) [reply pkg/apis/scope/v0alpha1/types.go:17]: Let me get back to you soon! :)

**Comment by bergquist** [comment]
Added some fields. Anything else that needs to be done?

**Comment by kylebrandt** [comment]
> Added some fields. Anything else that needs to be done?

Need to run generate (hack/update-codegen.sh) https://github.com/grafana/grafana/tree/main/hack , needs to be run from GOROOT/src/grafana/grafana , e.g. for me it is

```
# kbrandt @ kbrandt-ws in ~/go/github.com/grafana/grafana on git:main o [7:20:14] 
$ readlink -f .
/home/kbrandt/go/github.com/grafana/grafana

# kbrandt @ kbrandt-ws in ~/go/github.com/grafana/grafana on git:main o [7:20:18] 
$ GODEGEN_PKG=~/go/src/github.com/kubernetes/code-generator GOPATH=~/go ./hack/update-codegen.sh
```

**Comment by bergquist** [comment]
Done!

File: pkg/registry/apis/scope/register.go [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+func NewScopeAPIBuilder() *ScopeAPIBuilder {
+	return &ScopeAPIBuilder{}
+}
+
+func RegisterAPIService(features featuremgmt.FeatureToggles, apiregistration builder.APIRegistrar) *ScopeAPIBuilder {
+	if !features.IsEnabledGlobally(featuremgmt.FlagGrafanaAPIServerWithExperimentalAPIs) {
+		return nil // skip registration unless opting into experimental apis
+	}
+	builder := NewScopeAPIBuilder()
+	apiregistration.RegisterAPI(NewScopeAPIBuilder())
```
Comment (toddtreece) [inline pkg/registry/apis/scope/register.go:33]: should this be using the builder variable?
```suggestion
	apiregistration.RegisterAPI(builder)
```

**Review by toddtreece — APPROVED** [review]
one small comment, and i noticed the PR description is empty. might be worth adding some context there for future reference

**Pull request merged**